File: app/src/main/java/org/qp/android/questopiabundle/utils/StreamUtil.java
package org.qp.android.questopiabundle.utils;

import androidx.annotation.NonNull;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public final class StreamUtil {
    private static final int BUFFER_SIZE = 8192;

    /**
     * Copies data from the <code>from</code> stream to the <code>to</code> stream.
     */
    public static void copy(@NonNull InputStream from, OutputStream to) throws IOException {
        var buffer = new byte[BUFFER_SIZE];
        int bytesRead;
        while ((bytesRead = from.read(buffer)) > 0) {
            to.write(buffer, 0, bytesRead);
        }
    }
}

File: app/src/main/java/org/qp/android/questopiabundle/utils/StreamUtil.kt
package org.qp.android.questopiabundle.utils

import java.io.IOException
import java.io.InputStream
import java.io.OutputStream

object StreamUtil {
    private const val BUFFER_SIZE = 8192

    /**
     * Copies data from the `from` stream to the `to` stream.
     */
    @Throws(IOException::class)
    fun copy(from: InputStream, to: OutputStream) {
        val buffer = ByteArray(BUFFER_SIZE)
        var bytesRead: Int
        while ((from.read(buffer).also { bytesRead = it }) > 0) {
            to.write(buffer, 0, bytesRead)
        }
    }
}
